xmtp_mls/src/groups/mod.rs

```
@@ -132,6 +155,122 @@ where
+                let message = StoredGroupMessage {
+                    id: message_id,
+                    group_id: self.group_id.clone(),
+                    decrypted_message_bytes: message_bytes,
+                    sent_at_ns: envelope_timestamp_ns as i64, // TODO validate that this timestamp is controlled by server
+                    kind: GroupMessageKind::Application,
+                    sender_installation_id: sender_installation_id,
+                    sender_wallet_address: sender_account_address,
+                };
+                message.store(&mut self.client.store.conn()?)?;
+            }
+            ProcessedMessageContent::ProposalMessage(_proposal_ptr) => {
+                // intentionally left blank.
+            }
+            ProcessedMessageContent::ExternalJoinProposalMessage(_external_proposal_ptr) => {
+                // intentionally left blank.
+            }
+            ProcessedMessageContent::StagedCommitMessage(_commit_ptr) => {
+                // intentionally left blank.
+            }
+        }
+        Ok(())
+    }
+
+    pub fn process_messages(&self, envelopes: Vec<Envelope>) -> Result<(), GroupError> {
+        let provider = self.client.mls_provider();
+        let mut openmls_group = self.load_mls_group(&provider)?;
+        let receive_errors: Vec<MessageProcessingError> = envelopes
+            .into_iter()
+            .map(|envelope| -> Result<(), MessageProcessingError> {
+                let mls_message_in = MlsMessageIn::tls_deserialize_exact(&envelope.message)?;
+
+                match mls_message_in.extract() {
+                    MlsMessageInBody::PrivateMessage(message) => self.process_private_message(
+                        &mut openmls_group,
+                        &provider,
+                        message,
+                        envelope.timestamp_ns,
+                    ),
+                    _ => panic!("Unsupported message type"),
```

Should just return an error here and abort

---

I figure it might be better to loudly fail, seeing as this shouldn't ever happen? Might be good to know if we have any unexpected message types on the channel

---

Ah - if we leave this as-is then anyone can DOS a group by leaving an unsupported message type. Will change this